x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/pagerduty/PagerDutyService.java

```diff
@@ -45,7 +45,7 @@ public PagerDutyService(Settings settings, HttpClient httpClient, ClusterSetting
         clusterSettings.addAffixUpdateConsumer(SETTING_SERVICE_API_KEY, (s, o) -> {}, (s, o) -> {});
         clusterSettings.addAffixUpdateConsumer(SETTING_SECURE_SERVICE_API_KEY, (s, o) -> {}, (s, o) -> {});
         clusterSettings.addAffixUpdateConsumer(SETTING_DEFAULTS, (s, o) -> {}, (s, o) -> {});
-        setAccountSetting(settings);
+        loadSettings(settings);
```

Comment:
is this PagerDutyService missing  

```
clusterSettings.addSettingsUpdateConsumer(this::loadSettings, getSettings());
```

in its ctor?

Reply:
In order to prevent this mistake, you could move

```
clusterSettings.addSettingsUpdateConsumer(this::loadSettings, getSettings());
```

into the `NotificationService` constructor and pass the `getSettings()` as arg to `super()`

Reply:
I could also just have `NotificationService` impl a `getSettings` that the services overwrite. which is best?

Reply:
ahh, nm, its static, it would be more work to change all that.

Reply:
Im doing this in another commit, to ensure it does not muddy up this pull request. Ill edit this comment w/ the new PR once ive finished testing.

Reply:
#31762 has been merged